cpp/src/join/join_kernels.cuh

```diff
@@ -49,6 +49,19 @@ __inline__ __device__ void add_pair_to_cache(const size_type first,
   joined_shared_r[my_current_idx] = second;
 }
 
+/**
+ * @brief Remaps a hash value to a new value if it is equal to the specified sentinel value.
+ *
+ * @param hash The hash value to potentially remap
+ * @param sentinel The reserved value
+ */
+template <typename H, typename S>
+constexpr auto remap_sentinel_hash(H hash, S sentinel)
+{
+  // Arbitrarily choose hash - 1
+  return (hash == sentinel) ? --hash : hash;
```

Comment:
```suggestion
  return (hash == sentinel) ? hash - 1 : hash;
```
I think this is clearer... Especially since H is a template.

Reply:
What does the template have to do with it?

Reply:
You never know if it is a reference type or it implements an operator-- with side effects.

Using - avoids the risk of modifying the input.